File: src/main/java/de/spries/fleetcommander/model/core/universe/HasCoordinates.java
package de.spries.fleetcommander.model.core.universe;

public interface HasCoordinates {

    int getX();

    int getY();

    default double distanceTo(HasCoordinates other) {
        int distanceX = getX() - other.getX();
        int distanceY = getY() - other.getY();
        return Math.sqrt(Math.pow(distanceX, 2) + Math.pow(distanceY, 2));
    }
}

File: src/main/kotlin/de/spries/fleetcommander/model/core/universe/HasCoordinates.kt
package de.spries.fleetcommander.model.core.universe

import kotlin.math.pow
import kotlin.math.sqrt

open class HasCoordinates(var x: Int, var y: Int) {

    fun distanceTo(other: HasCoordinates): Double {
        val distanceX = x - other.x
        val distanceY = y - other.y
        return sqrt(distanceX.toDouble().pow(2.0) + distanceY.toDouble().pow(2.0))
    }
}
